tests/test_docs_serve.py

```
@@ -0,0 +1,135 @@
+
+def test_checkout_path_discovers_sibling(tmp_path: Path) -> None:
+    logfire_root = tmp_path / 'logfire'
+    sibling = tmp_path / 'unified-docs'
+    logfire_root.mkdir()
+    sibling.mkdir()
+
+    assert docs_serve._checkout_path(logfire_root, {}) == sibling
+
+
+def test_checkout_path_explains_private_preview_requirement(tmp_path: Path) -> None:
+    with pytest.raises(docs_serve._DocsServeError, match='requires access.*UNIFIED_DOCS_PATH'):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)